packages/@datapull/aws-origin/index.js

```diff
@@ -49,16 +49,94 @@ class AwsOrigin {
     console.log('[AWS Origin] params', params);
 
     return new Promise((resolve, reject) => {
-      serviceClient[this.config.method].call(serviceClient, params, (err, resp) => {
+      serviceClient[this.config.method].call(serviceClient, params, async (err, resp) => {
         if (err) {
           console.error('[AWS Origin] ERROR ', err);
           return reject(err);
         }
 
-        resolve(resp);
+        if (!this.config.awsDetailsCall) {
+          return resolve(resp);
+        }
+
+        // For every item in the response, make a details call:
+        try {
+          const details = await this.pullDataFromDetailsCall(this.config.awsDetailsCall, config, resp);
+          return resolve(details);
+        } catch (err) {
+          reject(err);
+        }
       });
     });
   }
+
+  async pullDataFromDetailsCall(detailsCallConfig, clientConfig, listCallResponse) {
+    // validate the config:
+    if (!detailsCallConfig.itemsListKey) {
```

Comment:
The following 4 checks are so repetitive I feel like it could be organized in a loop. But it's up to you.

Reply:
i'll keep it this way